### PR TITLE
[FIR] Introduce to getter and setter checks

### PR DESCRIPTION
**Pull request by yantimirov-timur**
Add GETTER_VISIBILITY_DIFFERS_FROM_PROPERTY_VISIBILITY and WRONG_SETTER_RETURN_TYPE

### CODE REVIEW

File: compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/jvm/FirGetterSetterChecker.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors.WRONG_SETTER_RETURN_TYPE
+import org.jetbrains.kotlin.fir.analysis.diagnostics.reportOn
+import org.jetbrains.kotlin.fir.declarations.FirProperty
+import org.jetbrains.kotlin.fir.declarations.visibility
+import org.jetbrains.kotlin.fir.types.ConeClassErrorType
+import org.jetbrains.kotlin.fir.types.coneType
+import org.jetbrains.kotlin.fir.types.isUnit
+
+
+object FirGetterSetterChecker : FirPropertyChecker() {
```
Comment (mglukhikh) [inline compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/jvm/FirGetterSetterChecker.kt:20]: All this code should be in `FirPropertyAccessorChecker` (will fix myself)

**Comment by mglukhikh** [comment]
"Generate FIR checker components and FIR/IDE diagnostics" wasn't run (will do myself).

File: compiler/testData/diagnostics/tests/classLiteral/smartCast.fir.kt [hunk 2]
```diff
@@ -17,7 +17,7 @@ fun test2(): KClass<out Foo>? {
     var f: Foo? = null
     if (f != null) {
         run { f = null }
-        return <!RETURN_TYPE_MISMATCH!><!EXPRESSION_OF_NULLABLE_TYPE_IN_CLASS_LITERAL_LHS!>f<!>::class<!>
+        return <!EXPRESSION_OF_NULLABLE_TYPE_IN_CLASS_LITERAL_LHS!>f<!>::class
```
Comment (mglukhikh) [inline compiler/testData/diagnostics/tests/classLiteral/smartCast.fir.kt:20]: I don't understand how your checker could provoke this change (will recheck myself)

File: compiler/testData/diagnostics/tests/Basic.fir.kt [hunk 3]
```diff
@@ -6,6 +6,11 @@ fun test() : Int {
     val a : () -> Unit = {
         foo(<!ARGUMENT_TYPE_MISMATCH!>1<!>)
     }
+
```
Comment (mglukhikh) [inline compiler/testData/diagnostics/tests/Basic.fir.kt:9]: Why this change is needed?

**Review by Sifat1998 — APPROVED** [review]

**Review by Sifat1998 — APPROVED** [review]

**Comment by mglukhikh** [comment]
Merged in the following form: https://github.com/JetBrains/kotlin/commit/1dfc3c0520fe6ae977e12c87eb5f9e43892b70d6
